gms/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java

```diff
@@ -59,26 +75,49 @@ protected RestHighLevelClient createInstance() {
   private static RestClientBuilder loadRestHttpClient(@Nonnull String host, int port, int threadCount,
       int connectionRequestTimeout) {
     RestClientBuilder builder = RestClient.builder(new HttpHost(host, port, "http"))
-        .setHttpClientConfigCallback(httpAsyncClientBuilder -> httpAsyncClientBuilder.setDefaultIOReactorConfig(
-            IOReactorConfig.custom().setIoThreadCount(threadCount).build()));
+        .setHttpClientConfigCallback(httpAsyncClientBuilder -> httpAsyncClientBuilder
+            .setDefaultIOReactorConfig(IOReactorConfig.custom().setIoThreadCount(threadCount).build()));
 
-    builder.setRequestConfigCallback(requestConfigBuilder -> requestConfigBuilder.
-        setConnectionRequestTimeout(connectionRequestTimeout));
+    builder.setRequestConfigCallback(
+        requestConfigBuilder -> requestConfigBuilder.setConnectionRequestTimeout(connectionRequestTimeout));
 
     return builder;
   }
 
   @Nonnull
   private static RestClientBuilder loadRestHttpsClient(@Nonnull String host, int port, int threadCount,
       int connectionRequestTimeout, @Nonnull SSLContext sslContext) {
+
     final RestClientBuilder builder = RestClient.builder(new HttpHost(host, port, "https"))
         .setHttpClientConfigCallback(httpAsyncClientBuilder -> httpAsyncClientBuilder.setSSLContext(sslContext)
             .setSSLHostnameVerifier(new NoopHostnameVerifier())
             .setDefaultIOReactorConfig(IOReactorConfig.custom().setIoThreadCount(threadCount).build()));
 
-    builder.setRequestConfigCallback(requestConfigBuilder -> requestConfigBuilder.
-        setConnectionRequestTimeout(connectionRequestTimeout));
+    builder.setRequestConfigCallback(
+        requestConfigBuilder -> requestConfigBuilder.setConnectionRequestTimeout(connectionRequestTimeout));
 
     return builder;
   }
+
+  @Nonnull
+  private static RestClientBuilder loadBasicAuthRestHttpsClient(@Nonnull String host, int port, int threadCount,
+      int connectionRequestTimeout, String username, String password) {
+
+    final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+    credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(username, password));
+
+    RestClientBuilder builder = RestClient.builder(new HttpHost(host, port, "https"))
```

Comment:
OK, I cannot push, but basically it's all there, you just need an SSL context anyway, since you are accessing https host: 
```java
    final RestClientBuilder builder = RestClient.builder(new HttpHost(host, port, "https"))
            .setHttpClientConfigCallback(
                    httpAsyncClientBuilder -> httpAsyncClientBuilder.setSSLContext(sslContext)
                            .setSSLHostnameVerifier(new NoopHostnameVerifier())//TODO: You would probably want to actually verify the hostnames
                            .setDefaultCredentialsProvider(credentialsProvider)
                            .setDefaultIOReactorConfig(IOReactorConfig.custom().setIoThreadCount(threadCount).build())
            );
```

This should work. 
Does it work? How exactly does it fail?